### DISCUSSION TITLE
incorrect transfer function of integrator

### DISCUSSION THREAD

**Discussion by ajitsk394** [mph-/lcapy#134] [Q&A] [answered]
I want to get the transfer function of an integrator. I wrote the schematics below
***************************************
from lcapy import Circuit, t, oo
a = Circuit("""
E 1 0 opamp 3 2 Ad; right, flipud
W 4 2; right
C 2_2 1_1 C 0; right
W 2 2_2; up
W 1 1_1; up
W 4 4_2; down=0.5
Is 4_2 0_3; down
W 0_3 0_1; down=0.5
W 3 0_2; down
#V 3 0_2 1; down
W 1 1_2; right
P 1_2 0; down
W 0_1 0_2; right
W 0_2 0; right
; draw_nodes=connections, label_ids=none, label_nodes=primary, scale=0.3
""")
a.draw()
***************************************
Vo = a[1].V
Vo.limit('Ad',oo)
***************************************
The result is proportional to 1/s^2 while it should only be proportional to 1/s.

Can anyone help me out here.

**Comment by mph-** [comment]
You have specified an initial condition for the capacitor so the circuit is solved as an initial value problem using Laplace transforms.  In this case your current source has a Laplace transform of Is / s so you are getting the step response.   What you want is the transimpedance of the circuit.  You can achieve this several ways, say by using a Dirac delta for the current source:

    Is 4_2 0_3 {delta(t)}; down

**Comment by mph-** [comment, marked as answer]
Oops, that should have been

    Is 4_2 0_3 {Is * delta(t)}; down

Alternatively, you can specify an arbitrary s-domain current:

    Is 4_2 0_3 {Is(s)}; down

**Comment by ajitsk394** [comment]
Thank you :)
